craft_providers/bases/buildd.py

```diff
@@ -0,0 +1,520 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Buildd image(s)."""
+import enum
+import io
+import logging
+import pathlib
+import subprocess
+import time
+from textwrap import dedent
+from time import sleep
+from typing import Dict, Optional
+
+from craft_providers import Base, Executor, errors
+
+from .errors import BaseConfigurationError
+
+logger = logging.getLogger(__name__)
+
+
+def default_command_environment() -> Dict[str, Optional[str]]:
+    """Provide default command environment dictionary.
+
+    The minimum environment for the buildd image to be configured and
+    function properly.
+
+    :returns: Dictionary of environment key/values.
+    """
+    return dict(
+        PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
```

Comment:
please add a comment explaining why this PATH

Reply:
done.